app/models/course/assessment.rb

```diff
@@ -233,9 +233,13 @@ def selected_test_type_for_grading
   # Re-grades all submissions to programming_questions after any change to
   # test case booleans has been committed
   def grade_with_new_test_cases
-    return if (previous_changes.keys & ['use_private', 'use_public', 'use_evaluation']).empty?
-    programming_questions.each do |question|
-      Course::Assessment::Question::AnswersEvaluationJob.perform_later(question)
+    return unless (previous_changes.keys & ['use_private', 'use_public', 'use_evaluation']).any? && autograded?
```

Comment:
I've been looking at this for some time and it's still quite hard to understand. How about creating a private function `regrade_programming_answers?` and moving the conditional there?

So this line will read `return unless regrade_programming_answers?`.

And you can add a descriptive comment at the `regrade_programming_answers?` definition that explains it's checking for changes to the booleans for autograded assessments.

Reply:
might be confusing if we pull this out because of the use of `previous_changes`. I can instead add a comment right above `return unless (previous_changes.keys & ['use_private', 'use_public', 'use_evaluation']).any? && autograded?` to explain what is does.

let me know you'd still prefer this check in a separate function.

Reply:
`previous_changes` is available in a separate function, no need to pass in as a param.